pkg/scheduler/api/validation/validation.go

```diff
@@ -28,16 +28,68 @@ import (
 	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"
 )
 
+// checkIfExists checks if the given name exists in the string slice.
+func checkIfExistsInList(name string, nameList []string) bool {
+	for _, itemName := range nameList {
+		if name == itemName {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
It's possible to make it a O(1) op (see https://github.com/kubernetes/kubernetes/pull/68103/commits/22242a072ef94ddac69771ff91fe9c5864f617d1#r214219724)